CriteoPublisherSdk/Sources/Logging/CR_RemoteLogHandler.m

```diff
@@ -18,11 +18,76 @@
 //
 
 #import "CR_RemoteLogHandler.h"
+#import "CR_RemoteLogStorage.h"
+#import "CR_RemoteLogRecord.h"
+
+@interface CR_RemoteLogHandler ()
+
+@property(nonatomic, readonly) CR_RemoteLogStorage *remoteLogStorage;
+
+@end
 
 @implementation CR_RemoteLogHandler
 
+- (instancetype)initWithRemoteLogStorage:(CR_RemoteLogStorage *)remoteLogStorage {
+  self = [super init];
+  if (self) {
+    _remoteLogStorage = remoteLogStorage;
+  }
+
+  return self;
+}
+
 - (void)logMessage:(CR_LogMessage *)message {
-  // TODO
+  // TODO EE-1374 check consent
+
+  // TODO check log level
+
+  CR_RemoteLogRecord *remoteLogRecord = [self remoteLogRecordFromLogMessage:message];
+  if (remoteLogRecord != nil) {
+    [self.remoteLogStorage pushRemoteLogRecord:remoteLogRecord];
+  }
+}
+
+- (CR_RemoteLogRecord *_Nullable)remoteLogRecordFromLogMessage:(CR_LogMessage *)logMessage {
+  NSString *message = [self messageBodyFromLogMessage:logMessage];
+  if (message == nil) {
+    return nil;
+  }
+
+  return [[CR_RemoteLogRecord alloc] initWithTag:logMessage.tag
+                                        severity:logMessage.severity
+                                         message:message
+                                   exceptionType:logMessage.exception.name];
+}
+
+- (NSString *_Nullable)messageBodyFromLogMessage:(CR_LogMessage *)logMessage {
+  if (logMessage.message.length == 0 && logMessage.exception == nil) {
+    return nil;
+  }
+
+  NSString *formattedDate = [self formatDate:logMessage.timestamp];
+
+  // A script to read the logs is sensible to the format: date should always be at the end of the
+  // message, separated with a ","
+  if (logMessage.exception) {
+    return [NSString
+        stringWithFormat:@"%@\n--- Exception: %@\n--- Stack: %@\n--- User info: %@,%@:%lu,%@",
+                         logMessage.message, logMessage.exception,
+                         logMessage.exception.callStackSymbols, logMessage.exception.userInfo,
+                         logMessage.file, (unsigned long)logMessage.line, formattedDate];
+  } else {
+    return [NSString stringWithFormat:@"%@,%@:%lu,%@", logMessage.message, logMessage.file,
+                                      (unsigned long)logMessage.line, formattedDate];
+  }
+}
+
+- (NSString *)formatDate:(NSDate *)date {
+  NSDateFormatter *formatter = NSDateFormatter.new;
```

Comment:
`NSDateFormatter` init is costly and it is recommended to cache them. I would suggest to keep this one as a static class property.

Reply:
Done